ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -251,20 +258,43 @@ protected Node visitJoin(final Join node, final AnalysisContext context) {
     return null;
   }
 
-  private String fetchKeyFieldName(Expression expression) {
+  private Pair<String, String> fetchKeyFieldName(ComparisonExpression comparisonExpression, String sourceAlias, Schema
```

Comment:
Not sure I follow the logic here. We are trying to match the alias with both the left and right side of the expression. Isn't this too inclusive? In the calling context we are fetching both the left and right side separately. If the alias matches either, won't this validation pass erroneously?

Reply:
This is to make sure that we don't impose an order on the comparison expression sides. For instance, left side can have the table and right side can have the stream key. We expect to see the key for both stream and key regardless of their place in the comparison expression.

Reply:
I see. Perhaps it would be worth adding a comment about this. Something like 'This method picks the side of the expression which matches the given alias'. 

An alternative would be to send the left and right side separately with both aliases, and allow either alias to be picked. Both are equivalent, but the suggested comment on the behavior would help, IMO.

Reply:
Added comments providing more details on the method.